src/Runner.Common/ProcessChannel.cs

```diff
@@ -96,5 +98,21 @@ private void Dispose(bool disposing)
                 _outClient?.Dispose();
             }
         }
+
+        //  Create a deterministic hash for a string to ensure that the message is not malformed on delivery
+        private static string GetMessageHash(string message)
+        {
+            var md5 = System.Security.Cryptography.MD5.Create();
+            byte[] input = System.Text.Encoding.UTF8.GetBytes(message);
+            byte[] hash = md5.ComputeHash(System.Text.Encoding.UTF8.GetBytes(message));
```

Comment:
@TingluoHuang do you know whether this will this flag security scans ? do we need to use sha256 or something?

Reply:
for fips compliance

Reply:
we should avoid using MD5, sha256 is better.